src/doom/d_main.c

```diff
@@ -1098,25 +1117,107 @@ static void InitGameVersion(void)
-    {
-        deathmatch = 1;
-    }
-    
-    // The original exe does not support retail - 4th episode not supported
+    // Version-specific differences
+    // Each entry is the last version where the statement is true.
+    switch (gameversion)
+    {
+        case exe_doom_1_0:
+            // Different trig tables
+            finetangent = finetangent1_0;
+            finesine = finesine1_0;
+            finecosine = &finesine1_0[FINEANGLES/4];
+            tantoangle = tantoangle1_0;
+
+            // Skull keys still spawn in deathmatch
+            mobjinfo[MT_MISC7].flags &= ~MF_NOTDMATCH;
+            mobjinfo[MT_MISC8].flags &= ~MF_NOTDMATCH;
+            mobjinfo[MT_MISC9].flags &= ~MF_NOTDMATCH;
+
+        case exe_doom_1_1:
+            // Env suits count for item %
+            mobjinfo[MT_MISC14].flags |= MF_COUNTITEM;
+
+        case exe_doom_1_2:
+            // Boss spider is not fullbright when attacking
+            // FIXME: not sure about last version, true at least in v1.2
+            states[S_SPID_ATK1].frame &= ~FF_FULLBRIGHT;
+            states[S_SPID_ATK2].frame &= ~FF_FULLBRIGHT;
+            states[S_SPID_ATK3].frame &= ~FF_FULLBRIGHT;
+            states[S_SPID_ATK4].frame &= ~FF_FULLBRIGHT;
+
+            // Powerups are not fullbright
+            // FIXME: not sure about last version, true at least in v1.2
+            states[S_SOUL].frame &= ~FF_FULLBRIGHT;
+            states[S_SOUL2].frame &= ~FF_FULLBRIGHT;
+            states[S_SOUL3].frame &= ~FF_FULLBRIGHT;
+            states[S_SOUL4].frame &= ~FF_FULLBRIGHT;
+            states[S_SOUL5].frame &= ~FF_FULLBRIGHT;
+            states[S_SOUL6].frame &= ~FF_FULLBRIGHT;
+            states[S_PINV].frame &= ~FF_FULLBRIGHT;
+            states[S_PINV2].frame &= ~FF_FULLBRIGHT;
+            states[S_PINV3].frame &= ~FF_FULLBRIGHT;
+            states[S_PINV4].frame &= ~FF_FULLBRIGHT;
+            states[S_PSTR].frame &= ~FF_FULLBRIGHT;
+            states[S_PINS].frame &= ~FF_FULLBRIGHT;
+            states[S_PINS2].frame &= ~FF_FULLBRIGHT;
+            states[S_PINS3].frame &= ~FF_FULLBRIGHT;
+            states[S_PINS4].frame &= ~FF_FULLBRIGHT;
+            states[S_SUIT].frame &= ~FF_FULLBRIGHT;
+            states[S_PMAP].frame &= ~FF_FULLBRIGHT;
+            states[S_PMAP2].frame &= ~FF_FULLBRIGHT;
+            states[S_PMAP3].frame &= ~FF_FULLBRIGHT;
+            states[S_PMAP4].frame &= ~FF_FULLBRIGHT;
+            states[S_PMAP5].frame &= ~FF_FULLBRIGHT;
+            states[S_PMAP6].frame &= ~FF_FULLBRIGHT;
+
+            // Max health is 199
+            // FIXME: not sure about last version, true at least in v1.2
+            deh_max_health = 199;
+            deh_max_soulsphere = 199;
+
+        //case exe_doom_1_3:
+        //case exe_doom_1_25:
+            // Deathmatch 2.0 not supported
+            if (deathmatch == 2)
+            {
+                deathmatch = 0;
+                if (M_CheckParm ("-deathmatch"))
```

Comment:
It wouldn't be `deathmatch == 2` without the `-deathmatch` parameter, right?

Reply:
`-altdeath` still sets `deathmatch == 2` even without the `-deathmatch` parameter.  Since `-altdeath` didn't exist before Doom v1.4, the `-deathmatch` parameter determines `deathmatch`.